src/test/java/net/sf/jabref/logic/bibtex/BibEntryWriterTest.java

```diff
@@ -403,4 +403,83 @@ public void trimFieldContents() throws IOException {
 
         assertEquals(expected, actual);
     }
+
+    @Test
+    public void roundTripWithPrecedingCommentTest() throws IOException {
+        // @formatter:off
+        String bibtexEntry = "% Some random comment that should stay here" + Globals.NEWLINE +
+                "@Article{test," + Globals.NEWLINE +
+                "  Author                   = {Foo Bar}," + Globals.NEWLINE +
+                "  Journal                  = {International Journal of Something}," + Globals.NEWLINE +
+                "  Note                     = {some note}," + Globals.NEWLINE +
+                "  Number                   = {1}" + Globals.NEWLINE +
+                "}";
+        // @formatter:on
+
+        // read in bibtex string
+        ParserResult result = BibtexParser.parse(new StringReader(bibtexEntry));
+
+        Collection<BibEntry> entries = result.getDatabase().getEntries();
+        assertEquals(1, entries.size());
+
+        BibEntry entry = entries.iterator().next();
+        assertEquals("test", entry.getCiteKey());
```

Comment:
Since a test should only fail because of one reason (i.e. in the best case only one assert). This test should check if parsing -> writing yields the same string which was originally passed to it. Thus only the last assert is what you want. https://github.com/JabRef/jabref/pull/1471/files#diff-b4b7ca399d396eb3e62fc3416156c92aR437